tests/scripts/thread-cert/Cert_5_1_02_ChildAddressTimeout.py

```diff
@@ -102,56 +130,82 @@ def test(self):
-        sed_messages = self.simulator.get_messages_sent_by(SED)
-
-        # 1 - All
-        leader_messages.next_mle_message(mle.CommandType.ADVERTISEMENT)
-
-        router1_messages.next_mle_message(mle.CommandType.PARENT_REQUEST)
-        leader_messages.next_mle_message(mle.CommandType.PARENT_RESPONSE)
-
-        router1_messages.next_mle_message(mle.CommandType.CHILD_ID_REQUEST)
-        leader_messages.next_mle_message(mle.CommandType.CHILD_ID_RESPONSE)
-
-        msg = router1_messages.next_coap_message("0.02")
-        msg.assertCoapMessageRequestUriPath("/a/as")
-
-        msg = leader_messages.next_coap_message("2.04")
-
-        router1_messages.next_mle_message(mle.CommandType.ADVERTISEMENT)
-
-        ed_messages.next_mle_message(mle.CommandType.PARENT_REQUEST)
-        router1_messages.next_mle_message(mle.CommandType.PARENT_RESPONSE)
-        ed_messages.next_mle_message(mle.CommandType.CHILD_ID_REQUEST)
-        router1_messages.next_mle_message(mle.CommandType.CHILD_ID_RESPONSE)
-
-        sed_messages.next_mle_message(mle.CommandType.PARENT_REQUEST)
-        router1_messages.next_mle_message(mle.CommandType.PARENT_RESPONSE)
-        sed_messages.next_mle_message(mle.CommandType.CHILD_ID_REQUEST)
-        router1_messages.next_mle_message(mle.CommandType.CHILD_ID_RESPONSE)
-
-        # 3 - Leader
-        msg = leader_messages.next_coap_message("0.02")
-        msg.assertCoapMessageRequestUriPath("/a/aq")
-
-        msg = leader_messages.next_coap_message("0.02")
-        msg.assertCoapMessageRequestUriPath("/a/aq")
-
-        # 4 - Router1
-        msg = router1_messages.does_not_contain_coap_message()
-
-        # 6 - Leader
-        msg = leader_messages.next_coap_message("0.02")
-        msg.assertCoapMessageRequestUriPath("/a/aq")
-
-        msg = leader_messages.next_coap_message("0.02")
-        msg.assertCoapMessageRequestUriPath("/a/aq")
-
-        # 7 - Router1
-        msg = router1_messages.does_not_contain_coap_message()
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER']
+        LEADER_RLOC = pv.vars['LEADER_RLOC']
+        ROUTER = pv.vars['ROUTER']
+        MED = pv.vars['MED']
+        MED_RLOC = pv.vars['MED_RLOC']
+        SED = pv.vars['SED']
+        SED_RLOC = pv.vars['SED_RLOC']
+
+        # Step 1: Verify topology is formed correctly
+
+        pv.verify_attached('ROUTER')
+
+        pkts.filter_wpan_src64(ROUTER).\
+            filter_wpan_dst64(MED).\
+            filter_mle_cmd(MLE_CHILD_ID_RESPONSE).\
+            must_next()
+
+        pkts.filter_wpan_src64(ROUTER).\
+            filter_wpan_dst64(SED).\
+            filter_mle_cmd(MLE_CHILD_ID_RESPONSE).\
+            must_next()
+
+        # Step 2: Power off both devices and allow for the keep-alive timeout to expire
+        # Step 3: The Leader sends an ICMPv6 Echo Request to MED and attempts to perform
+        #         address resolution by sending an Address Query Request
```

Comment:
one note that there will be no ping request sent out for mleid dest because there is no address notification.